src/main/java/fi/aalto/cs/apluscourses/utils/CourseHiddenElements.java

```diff
@@ -0,0 +1,113 @@
+package fi.aalto.cs.apluscourses.utils;
+
+import fi.aalto.cs.apluscourses.model.MalformedCourseConfigurationException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class CourseHiddenElements {
+
+  private final @NotNull List<Long> hiddenIDs = new ArrayList<>();
+
+  private final @NotNull List<Pattern> hiddenRegexes = new ArrayList<>();
+
+  private final @NotNull Map<String, List<Pattern>> hiddenLanguageSpecificRegexes = new HashMap<>();
```

Comment:
I'd make these subclasses of a common superclass/interface: one that filters by id, one that filters by regex, etc.

There could also be composite implementations of the same superclass: one with AND logic and the other with OR logic.

What do you think?

Reply:
I might agree with moving these three filtering features into separate classes. However, I'm skeptical of the value added by AND and OR logic:
- the OR logic can be achieved by just adding another filter
- the AND logic doesn't make much sense with the ID filter (since it can only match one or zero entries) so it could only be useful with the regex filters. And the regex language already has support for logical expressions anyway

Reply:
I feel like the AND- and OR-filters are somewhat of an overengineering for a feature that will almost invariably be used to "just hide one feedback exercise with substring X"